contracts/BurnableStakeBank.sol

```diff
@@ -60,52 +63,48 @@ contract BurnableStakeBank is IBurnableStakeBank, Lockable {
      * @param amount Amount of tokens to stake.
      * @param __data Data field used for signalling in more complex staking applications. //stakeLockBlockInterval
      */
-    function stakeFor(address user, uint256 amount, bytes __data) public onlyWhenUnlocked onlyWhenStakeInterval {
+    function stakeFor(address user, uint256 amount, bytes32 __data) public onlyWhenUnlocked onlyWhenStakeInterval {
         updateCheckpointAtNow(stakesFor[user], amount, false);
         updateCheckpointAtNow(stakeHistory, amount, false);
+        BurnableERC20 token = BurnableERC20( whitelist.getAddress(__data) );
```

Comment:
I generally try to program to an interface so I like doing:

IBurnableERC20 token = BurnableERC20(...); as it makes it easier to refactor later. Say if you wanted to pull local allocation from the function to be a private field later.